lib/hammer_cli_foreman/report_template.rb

```diff
@@ -77,6 +77,85 @@ def request_params
       build_options
     end
 
+    class ScheduleCommand < HammerCLIForeman::CreateCommand
+      command_name "schedule"
+      action :schedule_report
+
+      option '--inputs', 'INPUTS', N_('Specify inputs'),
+        :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--wait', :flag, _('Turns a command to be active, wait for the result and download it right away')
+      option '--path', "PATH", _("Path to directory where downloaded content will be saved. Only usable if wait is specified"),
+        :attribute_name => :option_path
+
+      def request_params
+        params = super
+        params['input_values'] = option_inputs || {}
+        params
+      end
+
+      def execute
+        data = send_request
+        if option_wait?
+          poll_for_report(data)
+        else
+          puts data['job_id']
+          HammerCLI::EX_OK
+        end
+      end
+
+      build_options
+
+      private
+
+      def poll_for_report(schedule_data)
+        report_data_args = build_report_data_args(schedule_data)
+        report_command = ReportDataCommand.new(invocation_path, context)
+        report_command.parse(report_data_args)
+        begin
+          response = report_command.send_request
+        end while response && response.code == 204
+        report_command.handle_success(response)
+      end
+
+      def build_report_data_args(schedule_data)
+        [
+          '--id', option_id,
+          '--job-id', schedule_data['job_id'],
+          '--path', option_path
+        ]
+      end
+    end
+
+    class ReportDataCommand < HammerCLIForeman::DownloadCommand
+      command_name "report-data"
+      action :report_data
+
+      def default_filename
+        "Report-#{Time.new.strftime("%Y-%m-%d")}.txt"
+      end
+
+      build_options
+
+      def execute
+        response = send_request
+        if response.code == 204
+          print_message(_('The report is not ready yet.'))
+          return HammerCLI::EX_OK
```

Comment:
nitpick: you can lower this statement to the end of the method and remove `return`.

Reply:
:dart: :+1: